### PR TITLE
Add File360 connector

### PR DESCRIPTION
**Pull request by SkNuwanTissera**
# Description

Add File360 connector

Related 

One line release note: 
- Add File360 connector

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

**Test Configuration**:
* Ballerina Version: Ballerina Swan Lake beta 3
* Operating System: Ubuntu
* Java SDK: Java 11

# Checklist:

### Changes to OpenAPI definition

- [X] Modified `info` section of the OpenAPI definition file - [Example](https://github.com/ballerina-platform/ballerina-extended-library/discussions/74)

    - `description` - what the connector is about 
    - `x-ballerina-init-description` OpenAPI extension - connector initialization details 
    - `x-ballerina-display` OpenAPI extension - connector name and path to the connector icon 

- [X] Added proper description each API Key, if API Key authentication is defined in OpenAPI definition file 

### Changes to connector module

- [X] Added license header at the top of each .bal file. 
- [X] Added Package.md as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/77)
- [X] Added Module.md for the module of the  connector as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/78)
- [ ] Created a directory called `resources` at the root of the connector and copied the connector icon there. 
      Name of the icon file need to be `<connector-name>.svg`
- [X] Added Ballerina.toml file with following information. For keywords refer to [guide](https://github.com/ballerina-platform/ballerina-extended-library/discussions/72)

### Verifying connector module

- [X] Compiled the connector successfully with the targeted ballerina version. 
- [X] Conducted smoke tests on the connector (Recommended when OpenAPI definition is obtain from an unofficial source)

### Security checks
 - [X] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [X] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

**Review by sachinira — APPROVED** [review]
Seems OK. But have one concern regarding the authentication mechanism of the operations. The current code goes along with the API doc.

**Review by LakshanSS — APPROVED** [review]

File: openapi/file360/Ballerina.toml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[package]
+license = ["Apache-2.0"]
+keywords = ["Content & Files/Documents", "Cost/Freemium"]
+org = "ballerinax"
+name = "file360"
+repository = "https://github.com/ballerina-platform/ballerinax-openapi-connectors"
+version = "1.0.0-SNAPSHOT"
```
Comment (SkNuwanTissera) [inline openapi/file360/Ballerina.toml:7]:
```suggestion
version = "1.0.0"
```

File: openapi/file360/Ballerina.toml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+[package]
+license = ["Apache-2.0"]
+keywords = ["Content & Files/Documents", "Cost/Freemium"]
+org = "ballerinax"
+name = "file360"
```
Comment (RolandHewage) [inline openapi/file360/Ballerina.toml:5]: please add the following line
icon = "icon.png"

File: openapi/file360/client.bal [hunk 3]
```diff
@@ -0,0 +1,826 @@
...
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+
+# This is a generated connector from [file360](https://developer.opentext.com/apis/ebc5860f-3e04-4d1b-a8be-b2683738c701/File360) OpenAPI Specification.
+# An API that allows interaction with File360 Web client.
+# The File360 Web client, a world class user experience powered by Angular/CSS, is built upon a set of RESTful APIs. The power of the web application is available to our customers and partners with the RESTful APIs which power every bit of the web application.
+# Please see https://file360demo.otxlab.com/File360Web/test/api for examples and test harness.
+@display {label: "file360", iconPath: "resources/file360.svg"}
```
Comment (RolandHewage) [inline openapi/file360/client.bal:23]: replace iconPath with 
iconPath: "icon.png"

File: openapi/file360/openapi.yml [hunk 4]
```diff
@@ -0,0 +1,4529 @@
...
+    url: http://support.opentext.com
+    email: support@opentext.com
+  version: "1.0"
+  x-ballerina-init-description: >
+    The connector initialization requires setting the API credentials. 
+    
+    Create an [OpenText account](https://login.opentext.com/register) and obtain an API key.
+  x-ballerina-display:
+    label: file360
+    iconPath: "resources/file360.svg"
```
Comment (RolandHewage) [inline openapi/file360/openapi.yml:24]: Replace iconPath with
iconPath: "icon.png"

**Comment by RolandHewage** [comment]
Please add the icon, so that this PR can be merged

File: openapi/file360/Ballerina.toml (hunk 2 above)
Comment (SkNuwanTissera) [reply openapi/file360/Ballerina.toml:5]: Added as sugested.

File: openapi/file360/Ballerina.toml (hunk 1 above)
Comment (SkNuwanTissera) [reply openapi/file360/Ballerina.toml]: Added as sugested.

File: openapi/file360/openapi.yml (hunk 4 above)
Comment (SkNuwanTissera) [reply openapi/file360/openapi.yml]: Added as suggested.

File: openapi/file360/client.bal (hunk 3 above)
Comment (SkNuwanTissera) [reply openapi/file360/client.bal]: Added as suggested.

**Pull request merged**